src/relay/qnn/op/dequantize.cc

```diff
@@ -65,8 +65,9 @@ bool DequantizeRel(const Array<Type>& types, int num_inputs, const Attrs& attrs,
     ICHECK_GE(axis, 0) << "axis " << dequantize_attrs->axis << " is out of range";
   }
 
+  // We assume per-channel dequantization if rank > 1, otherwise it's per tensor.
   PrimExpr axis_shape;
-  if (rank > 0) {
+  if ((!(scale_is_scalar && zp_is_scalar)) && (rank > 1)) {
```

Comment:
Hmm I believe we can still have per channel quantization if rank == 1. In this case it would just be every element has it's own set of quantization params. 

I think the proper check is to change `*_is_scalar` variables to see if either the rank of `scale` or `zp` is either of rank 0, or of arbitrary rank but with one element.